src/util/strencodings.cpp

```diff
@@ -81,16 +81,21 @@ template <typename Byte>
 std::optional<std::vector<Byte>> TryParseHex(std::string_view str)
 {
     std::vector<Byte> vch;
```

Comment:
Ideally (when no spaces present) two characters will form a single byte - and when spaces are present, it will be less, so we can avoid reallocation by reserving `str.size() / 2`

Reply:
nit: may be unnecessary / obvious, but an inline comment saying // two hex characters form a single byte will be more helpful to readers of the code than the comment you've added above.

Reply:
Done, thanks